src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -7476,6 +7480,23 @@ export class StudyViewPageStore {
         default: {},
     });
 
+    private calculateSampleCount(
+        result: (SampleTreatmentRow | PatientTreatmentRow)[] | undefined
+    ) {
+        if (!result) {
+            return 0;
+        }
+
+        return result
```

Comment:
this is confusing.  look like you are just returning an integer count, so why then bother populating allSamples with ids?   and why return size as any?

Reply:
Because a patient or sample can receive multiple treatments, it is possible for the same unique ID to appear in multiple rows. We want to count the number of unique samples being considered by this table, not the summation of each row's sample count. In order to do this, we have to construct a unique collection of sample ids and then return its count, rather than simply counting.

You're right about the `any` cast. I'll fix that.